subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -667,6 +671,56 @@ static void dhcpv4_timeout(struct k_work *work)
 	}
 }
 
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS_VENDOR_SPECIFIC)
+
+static int dhcpv4_parse_option_vendor(struct net_pkt *pkt, struct net_if *iface,
+				      enum net_dhcpv4_msg_type *msg_type, uint8_t length)
+{
+	struct net_dhcpv4_option_callback *vs_cb, *tmp;
+	struct net_pkt_cursor vs_backup;
+	uint8_t len;
+	int len_counter = length;
+	uint8_t vs_type;
+
+	while (!net_pkt_read_u8(pkt, &vs_type) && len_counter > 0) {
+		if (vs_type == DHCPV4_OPTIONS_END) {
+			NET_DBG("Vendor-specific options_end");
```

Comment:
Shouldn't we check if `len_counter` (`length`) is 0 here? Otherwise, it means END marker was read too early (malformed option?)